android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt

```diff
@@ -298,4 +302,28 @@ open class AppMainActivity : BaseMultiLanguageActivity(), QuestionnaireHandler,
       }
     }
   }
+
+  private fun overrideOnBackPressListener() {
+    onBackPressedDispatcher.addCallback(
+      object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() {
+          val title =
+            appMainViewModel.applicationConfiguration.appExitDialog?.title
+              ?: getString(R.string.exit_app)
+          val message =
+            appMainViewModel.applicationConfiguration.appExitDialog?.message
+              ?: getString(R.string.exit_app_message)
+          AlertDialogue.showAlert(
+            this@AppMainActivity,
+            alertIntent = AlertIntent.CONFIRM,
+            title = title,
+            message = message,
+            cancellable = false,
+            confirmButtonListener = { finish() },
+            neutralButtonListener = { dialog -> dialog.dismiss() },
```

Comment:
@hamza-vd This alert should only be shown if there is no more screen to navigate back to. Can you confirm that is the case based on your implementation? I assume that every time the user clicks the back button then, they will be prompted via the dialog (according to my understanding of your code)

Reply:
This does thanks. @f-odhiambo ^